docs/generators/MSSqlDDL.md

```diff
@@ -0,0 +1,494 @@
+The TmStamp field can automatically be added to the table by specifying the `--flag "add timestamp"` flag:
+```
+TABLE Example
+    ID                  SEQUENCE
+    SOME_DATA           CHAR(100)
+    USER_ID             USERSTAMP
+```
+Jportal Params:
+```
+--builtin-generator MSSqlDDL:\tmp --flag "add timestamp"
+```
+Example output:
+```
+CREATE TABLE ExampleSchema.Example
+(
+  ID INTEGER IDENTITY(1,1) NOT NULL
+, SOME_DATA VARCHAR(100) NOT NULL
+, USER_ID VARCHAR(50) NOT NULL
+, TIMESTAMP
+, CONSTRAINT PK_ExampleSchema_Example_PKEY PRIMARY KEY (
+    ID
+  )
+)
+GO
+```
+
+## UserId and TmStamp
+UserId's and TmStamp fields can automatically be added to the table by specifing `--flag "internal stamps"` to the generator.
+```
+TABLE Example
+    ID                  SEQUENCE
+    SOME_DATA           CHAR(100)
+```
+Jportal Params:
+```
+--builtin-generator MSSqlDDL:\tmp --flag "internal stamps"
+```
+Example output:
+```
+CREATE TABLE ExampleSchema.Example
+(
+  ID INTEGER IDENTITY(1,1) NOT NULL
+, SOME_DATA VARCHAR(100) NOT NULL
+, UpdateWhen  DATETIME DEFAULT CURRENT_TIMESTAMP NULL
+, UpdateByWho CHAR(8)  DEFAULT USER NULL 
+, CONSTRAINT PK_ExampleSchema_Example_PKEY PRIMARY KEY (
+    ID
+  )
+)
+GO
+```
+
+# Triggers
+## Insert trigger
+An insert trigger can be used to automatically set `ID`, `TIMESTAMP` and `USERSTAMP` fields. In order for this to generate in the DDL you need to specify the Primary key of type `SEQUENCE` or `BIGSEQUENCE` and pass the `--flag "use insert trigger"` to the generator:
+
+Example si file:
+```
+TABLE Example
+    ID                  SEQUENCE
+    SOME_DATA           CHAR(100)
+    USER_ID             USERSTAMP
+    TMSTAMP             TIMESTAMP
+```
+Jportal Params:
+```
+--builtin-generator MSSqlDDL:\tmp --flag "use insert trigger"
+```
+
+This will generate the DDL:
+```
+CREATE TABLE ExampleSchema.Example
+(
+  ID INTEGER NOT NULL
+, SOME_DATA VARCHAR(100) NOT NULL
+, USER_ID VARCHAR(50) NOT NULL
+, TMSTAMP DATETIME NOT NULL
+, CONSTRAINT PK_ExampleSchema_Example_PKEY PRIMARY KEY (
+    ID
+  )
+)
+GO
+
+IF OBJECT_ID('ExampleSchema.ExampleInsertTrigger','TR') IS NOT NULL
+    DROP TRIGGER ExampleSchema.ExampleInsertTrigger
+GO
+
+CREATE TRIGGER ExampleSchema.ExampleInsertTrigger ON ExampleSchema.Example FOR INSERT AS
+UPDATE ExampleSchema.Example SET ID=ID+0
+WHERE ID=(SELECT MAX(ID) FROM ExampleSchema.Example)
+UPDATE ExampleSchema.Example
+SET
+  ID = (SELECT MAX(ID) FROM ExampleSchema.Example)+1
+, USER_ID = USER_NAME()
+, TMSTAMP = GETDATE()
+WHERE ID = (SELECT ID FROM INSERTED)
+GO
+```
+
+
+# Audit Trigger
+An Audit trigger can be generated to track `INSERT`, `DELETE` and `UPDATE` queries performed on the table. For this to happed the flag `--flag "audit triggers"` should be passed to the generator. If the flag is passed a secondry table will be generated in the DDL called `<SCHEMA><TABLENAME>Audit`. When an `INSERT`, `UPDATE` or `DELETE` query is executed this table will be inserted into with:
```

Comment:
New, existing/undocumented?

Reply:
Does it occur in all the generators?

Reply:
No. Only MSSqlDDL has this Audit functionality.